test/onFailure.js

```diff
@@ -0,0 +1,13 @@
+//Note:this test currently fails
+var tap = require("tap");
+var tape = require("../");
+const t = tap.test("on failure", {timeout: 1000}, function (tt) {
+    tt.plan(1);
+    tape.onFailure(function() {
```

Comment:
could this add an `onFinish` listener as well that fails?

Reply:
@ljharb, i don't quite understand your comment.
This test does, in fact, fail. That is, for the "outer test" to pass, the "inner test" must fail, and this failure propagates. I wonder if you, or anyone else might have an idea about how to get around this?

Reply:
Sorry, what I mean is, add something like `tape.onFinish(function () { tt.fail(); })`

Reply:
@ljharb I couldn't quite get the method that you mentioned to work, though I was able to get a test working after a while. Hopefully that does it. Thanks again for the help and feedback!